core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt

```diff
@@ -156,18 +157,58 @@ object FileUtils {
     return null
   }
 
-  private fun documentProviderContentQuery(context: Context, uri: Uri) =
-    contentQuery(
-      context,
-      ContentUris.withAppendedId(
-        Uri.parse("content://downloads/public_downloads"),
-        try {
-          DocumentsContract.getDocumentId(uri).toLong()
-        } catch (ignore: NumberFormatException) {
-          0L
-        }
-      )
+  private fun documentProviderContentQuery(context: Context, uri: Uri): String? {
+    // Extracting the document ID from the URI.
+    val documentId = extractDocumentId(uri)
```

Comment:
We can test this function for all the content URI's provided.

Reply:
I have added the test cases for both `extractDocumentId` and `documentProviderContentQuery` methods for testing all content URIs.